### ISSUE TITLE
New IITC-Mobile UI(?)

### ISSUE THREAD

**Issue by modos189** [IITC-CE/ingress-intel-total-conversion#171]
I decided to play around with design and got this.

1. Menu is moved to the bottom of the screen for easy pressing with one hand

2. Map under status bar

3. Button to open chat and notifications (+ tabs)

Free space in left sliding block, I want to move there some of the most common settings and plug-in settings. At the moment, the left block contains settings only for a very limited number of plugins.

<details>
<summary>Screenshots</summary>

![photo_2019-04-23_13-04-18](https://user-images.githubusercontent.com/1785196/56573689-55df2f80-65ca-11e9-9d5a-667ecb87373f.jpg)
![photo_2019-04-23_13-05-19](https://user-images.githubusercontent.com/1785196/56573691-5677c600-65ca-11e9-9d11-96d68626d00f.jpg)

</details>

I would like to know how much you like this idea, whether it makes sense to develop it.
Also I would like to know ideas for improvements.

**Comment by johnd0e** [comment]
> 1. Menu is moved to the bottom of the screen for easy pressing with one hand
> 2. Map under status bar
> 3. Button to open chat and notifications (+ tabs)

In my opinion we always can do swipe with one hand.
And bottom part of screen  actually already have control: `div#mobileinfo` ('tap here for info screen'), which will become less accessible with proposed redesign.
(And in my own iitc setup right part `div#innerstatus` is also used).

> Free space in left sliding block, I want to move there some of the most common settings and plug-in settings. At the moment, the left block contains settings only for a very limited number of plugins.

I suppose it would be good.
But please consider to separate Tools (e.g. Poltal list) from Options (e.g. Draw-tools opt).
(I suppose we should separate them at desktop too)

**Comment by modos189** [comment]
It should be noted that swipe from bottom to open chat can not be done.

> I suppose we should separate them at desktop too

agreed

**Comment by johnd0e** [comment]
> It should be noted that swipe from bottom to open chat can not be done.

Right, it couldn't be directly invoked by swipe, as by swipe we open menu, from where we need to select chat tab.

About button: what of 3 chat tabs do you propose to open?

**Comment by johnd0e** [comment]
Button itself looks good, but I do not know where to place it..

Btw, it's easy to setup plugin to open chat on `div#mobileinfo` 'taphold' event.
At first we could open 'All' chat, and then remember last used tab.

P.S.
We also could setup 'taphold' listener on input box title in chat window, it'd cycle between tabs.

**Comment by johnd0e** [comment]
BTW, I like your horizontal chat tabs.

**Comment by modos189** [comment]
> Right, it couldn't be directly invoked by swipe, as by swipe we open menu, from where we need to select chat tab.

No, another reason. Chat opens inside WebView, but panel is part of Android UI. Theoretically, we can transfer information to application and draw messages using Android UI, but it makes little sense.

> what of 3 chat tabs do you propose to open

> At first we could open 'All' chat, and then remember last used tab.

yeap

**Comment by Suburbanno** [comment]
I really like the idea, the chat in tabs is perfect !! I would also like to keep my suggestion to the drawer: https://github.com/IITC-CE/ingress-intel-total-conversion/issues/139
and also a change in colors, it is currently green, and I love it for being an ENL player, but I think it should be a color for both factions, maybe a dark theme? could also give an improvement in chat conversations.

**Comment by Suburbanno** [comment]
@modos189 Do you have a test version ready for me to test? I'm really excited !!

**Comment by johnd0e** [comment]
> No, another reason. Chat opens inside WebView, but panel is part of Android UI. Theoretically, we can transfer information to application and draw messages using Android UI, but it makes little sense

Why would we need this?
In js we are able directly call `window.show('chat')`. 

I suppose that call js from Java is also elementary.

**Comment by johnd0e** [comment]
> and also a change in colors, it is currently green, and I love it for being an ENL player, but I think it should be a color for both factions, maybe a dark theme?

I'm thinking about using stock intel default color scheme.

@modos189 ?

**Comment by modos189** [comment]
> it is currently green

> I'm thinking about using stock intel default color scheme.

![изображение](https://user-images.githubusercontent.com/1785196/56645725-e54a1880-6686-11e9-8011-0c76ed5a0366.png)

It's dark-blue now. But if change, I prefer gray. Probably. It need to test.

> Do you have a test version ready for me to test?

No, for now this is a layout, even menu does not open.

> Chat opens inside WebView, but panel is part of Android UI.

> Why would we need this?

I mean, we can't open chat inside pull-out panel:

![photo_2019-04-23_13-04-18_f](https://user-images.githubusercontent.com/1785196/56648154-74592f80-668b-11e9-823b-7402176d25e4.jpg)

But clicking on button to open and close chat, of course, is possible.

**Comment by modos189** [comment]
I have prepared other versions of the layouts:

<details>
<summary>Details</summary>

![photo_2019-04-23_13-04-18_v2](https://user-images.githubusercontent.com/1785196/56649466-df0b6a80-668d-11e9-9e93-27924b64b6d8.jpg)
![photo_2019-04-23_13-04-18_v3](https://user-images.githubusercontent.com/1785196/56649468-df0b6a80-668d-11e9-93f3-5cdd0583afac.jpg)

</details>

If i choose between these two, I prefer the second one. Besides, I'm not sure I can implement the first one.

I just wanted the map to appear under the status bar, I think it's very modern)

**Comment by johnd0e** [comment]
> I mean, we can't open chat inside pull-out panel:

I see now. It would be great..

Is it possible to use swipe up as gesture only, to open usual chat pane?


> but panel is part of Android UI.

What if we open WebView there?
I wonder if it's possible to transfer data between webviews.

**Comment by modos189** [comment]
> Is it possible to use swipe up as gesture only, to open usual chat pane?

I think it's possible

**Comment by modos189** [comment]
> What if we open WebView there?
> I wonder if it's possible to transfer data between webviews.

In the second WebView, open a lightweight version of IITC that contains only chat? It can be done.

**Comment by Suburbanno** [comment]
@modos189 How's the design going?

**Comment by modos189** [comment]
@Suburbanno I haven't done anything yet. Besides, I don't know yet in which direction to develop UI.

**Comment by modos189** [comment]
![изображение](https://user-images.githubusercontent.com/1785196/58210113-24ed3a00-7cf1-11e9-9116-9d1041f22dd9.png)

maybe something like this

**Comment by Suburbanno** [comment]
@modos189 Well interesting what you did, but what do you think of using this? 
https://github.com/mikepenz/MaterialDrawer
really like the fact that I show account information on the drawer

**Comment by modos189** [comment]
![изображение](https://user-images.githubusercontent.com/1785196/58230707-c4381e80-7d3d-11e9-91c3-bb5d1ddd1bea.png)
I'm not sure what it's possible to do with standard Android tools

**Comment by Suburbanno** [comment]
omg, that's exactly what I was talking about, continue on this path and we'll have a UI/UX renewal

**Comment by johnd0e** [comment]
@modos189 
I like profile info in side drawer.
But not other entries:
- those separators look bad, it's not like material design
- too many mixed and unrelated items.
  Better leave plugin items only.
  Chat could be opened by swipe-from-bottom gesture.
  Misc settings deserve separate menu, like it is now (but we could move menu button in different place)

**Comment by Suburbanno** [comment]
no news? 😰

**Comment by modos189** [comment]
> Chat could be opened by swipe-from-bottom gesture.

Okay, let's say

> Misc settings deserve separate menu, like it is now (but we could move menu button in different place)

Where do put separate menu? The top panel will be overloaded with buttons

Anyway, we'd have an android developer. It seems that it will be difficult for me to implement.

**Comment by johnd0e** [comment]
> Where do put separate menu? 

For example, it could be represented by separate item in hamburger menu (near Settings),   and for easy access additional gesture, like hold on hamburger icon (or layers icon).

**Comment by johnd0e** [comment]
Let's talk about not so big redesign.
How about to transfer all links from info screen's toolbox to left side drawer?

Currently there are info, chats, and some plugins panes. But it'd be great to move all plugins there.
And as we have a lot plugins it'd be convenient to organize them into accordion menu.

**Comment by Suburbanno** [comment]
issue is dead? :(

**Comment by Suburbanno** [comment]
@modos189 could you let me know if the idea about the redesign of IITC Mobile was abandoned? something new in mind?

**Comment by modos189** [comment]
https://github.com/IITC-CE/IITC-Prime
